### DISCUSSION TITLE
browser shows CORS error when I enable proxy & CORS in setting

### DISCUSSION THREAD

**Discussion by qiulang** [mockoon/mockoon#436] [Help] [answered]
Please refer to the following 4 CORS errors. They are my mock api

<img width="1209" alt="CORS error" src="https://user-images.githubusercontent.com/516243/110415990-e1cea780-80cd-11eb-8601-628c1bc29abe.png">

This is my setting

![setting](https://user-images.githubusercontent.com/516243/110417020-bcdb3400-80cf-11eb-9a5b-bdfbfee54af5.png)

But if I check the logs, mockoon actually sent out response
![response](https://user-images.githubusercontent.com/516243/110416989-aa60fa80-80cf-11eb-8ae7-01abb20401b2.png)

**Comment by qiulang** [comment]
If I add CORS headers to environment headers, the mock apis work fines but the proxy api reports CORS error

<img width="1271" alt="proxy" src="https://user-images.githubusercontent.com/516243/110418067-c5346e80-80d1-11eb-8099-c6ccad54ef5c.png">

**Comment by qiulang** [comment, marked as answer]
OK I found the reason. Our server code requires "Access-Control-Allow-Headers" contains some customer value like following, after I add them the problem goes away.

`Access-Control-Allow-Headers:timestamp,source,nonce,token`

**Reply by 255kb** [reply]
Indeed, the CORS option only reply to the browser's OPTIONS pre-flight requests. To actually solve your CORS errors you also need to add the usual headers.
